packages/phone-number-privacy/src/match-making/get-contact-matches.ts

```diff
@@ -48,12 +49,15 @@ function isValidGetContactMatchesInput(requestBody: any): boolean {
   return (
     hasValidAccountParam(requestBody) &&
     hasValidUserPhoneNumberParam(requestBody) &&
-    hasValidContractPhoneNumbersParam(requestBody)
+    hasValidContractPhoneNumbersParam(requestBody) &&
+    isBodyReasonablySized(requestBody)
```

Comment:
what is this protected against? or validating for explicitly?

Reply:
Since we pass the request body (user input) into the EC recover function, I wanted to be sure that nothing the user gave us could break the function. I'm not sure how the hash calculation scales with the size of the input so I tried to set reasonable size restrictions.